lib/core/src/main/java/io/atlasmap/actions/StringComplexFieldActions.java

```diff
@@ -198,19 +199,16 @@ public static String padStringLeft(Action action, String input) {
         return builder.toString();
     }
 
-    @AtlasFieldActionInfo(name = "Prepend", sourceType = FieldType.STRING, targetType = FieldType.STRING, sourceCollectionType = CollectionType.NONE, targetCollectionType = CollectionType.NONE)
-    public static String prepend(Action action, Object input) {
-        if (!(action instanceof Prepend)) {
-            throw new IllegalArgumentException("Action must be a Prepend action");
-        }
-        String string = ((Prepend) action).getString();
-        if (input == null && string == null) {
-            return null;
+    @AtlasActionProcessor
+    public static String prepend(Prepend action, String input) {
+        String string = action.getString();
```

Comment:
This is the one action handler that I've ported over to using the `@AtlasActionProcessor` style, instead of the @AtlasFieldActionInfo style.  Notice that it uses the message signature to discover what Action it should be applied to and the type of input and output fields that it operates on.